FluentValidation.AutoValidation.Mvc/src/Filters/FluentValidationAutoValidationActionFilter.cs

```diff
@@ -55,15 +55,15 @@ public async Task OnActionExecutionAsync(ActionExecutingContext actionExecutingC
                     if (actionExecutingContext.ActionArguments.TryGetValue(parameter.Name, out var subject))
                     {
                         var parameterInfo = (parameter as ControllerParameterDescriptor)?.ParameterInfo;
-                        var parameterType = parameter.ParameterType;
+                        var parameterType = subject?.GetType();
                         var bindingSource = parameter.BindingInfo?.BindingSource;
 
                         var hasAutoValidateAlwaysAttribute = parameterInfo?.HasCustomAttribute<AutoValidateAlwaysAttribute>() ?? false;
                         var hasAutoValidateNeverAttribute = parameterInfo?.HasCustomAttribute<AutoValidateNeverAttribute>() ?? false;
 
                         if (subject != null && parameterType.IsCustomType() &&
                             !hasAutoValidateNeverAttribute && (hasAutoValidateAlwaysAttribute || HasValidBindingSource(bindingSource)) &&
-                            serviceProvider.GetValidator(parameterType) is IValidator validator)
+                            serviceProvider.GetValidator(parameterType!) is IValidator validator)
```

Comment:
@icnocop now you are overriding the nullability over here, I would do:

```
subject != null && parameterType != null && parameterType.IsCustomType() && etc...
```

Reply:
I'm not sure when `parameterType` would be null when `subject` is not null.

In any case, I just pushed another commit to support a potentially nullable `parameterType`, as you recommended.

Thank you.